src/Simplify.cpp

```diff
@@ -47,6 +47,38 @@ bool is_simple_const(Expr e) {
     return false;
 }
 
+// If the Expr is (var relop const) or (const relop var),
+// fill in the var name and return true.
+template<typename RelOp>
+bool is_var_relop_simple_const(Expr e, string* name) {
+    if (const RelOp *r = e.as<RelOp>()) {
+        if (is_simple_const(r->b)) {
+            const Variable *v = r->a.template as<Variable>();
+            if (v) {
+                *name = v->name;
+                return true;
+            }
+        }
+        else if (is_simple_const(r->a)) {
+            const Variable *v = r->b.template as<Variable>();
+            if (v) {
+                *name = v->name;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool is_var_simple_const_comparison(Expr e, string* name) {
```

Comment:
More generally, it's going to collapse if the var name matches, and the type of the comparison also matches, where the types are [<, <=], [==, !=], and [>, >=]

So maybe you could add another return parameter indicating which of the three types of comparison it is